doc/apiary/v2/fiware-ngsi-simple-cookbook.apib

```diff
@@ -0,0 +1,872 @@
+
++ Response 200 (application/json)
+
+        {
+            "type": "Room",
+            "id": "Bcn_Welt",
+            "temperature": 21.7,
+            "humidity": 60,
+            "location": {
+                "value":  "41.3763726, 2.1864475",
+                "type": "geo:point",
+                "crs": "WGS84"
+            }
+        }
+
+### Retrieve all entity data by Id. Not exists [GET /v2/entities/{entityId}]
+
++ Parameters
+    + entityId: Bcn_Welt (required, string) - Entity Id.
+
++ Response 404 (application/json)
+
+        {
+            "code": 4567,
+            "message": "Not Found",
+            "description": "There is no entity with such an id"
+        }
+
+### Retrieve all entity data by id and type [GET /v2/entities/type/{entityType}/id/{entityId}]
+
++ Parameters
+    + entityId: Boe_Idearium (required, string) - Entity ID
+    + entityType: Room (required, string) - Entity Type
+
++ Response 200 (application/json)
+
+        {
+            "type": "Room",
+            "id": "Boe_Idearium",
+            "temperature": 22.5
+        }
+
+### Get one entity attribute [GET /v2/entities/{entityId}{?attrs}]
+
++ Parameters
+    + entityId: Bcn_Welt (required, string) - Entity ID
+    + attrs: temperature (optional, string) - Attribute to be retrieved
+
++ Response 200 (application/json)
+
+        {
+            "temperature": 21.7
+        }
+
+### Get one entity attribute. Not present [GET /v2/entities/{entityId}{?attrs}]
+
++ Parameters
+    + entityId: Bcn_Welt (required, string) - Entity ID
+    + attrs: mistake (optional, string) - Attribute to be retrieved
+
++ Response 200 (application/json)
+
+        {
+        }
+
+### Get several entity attributes [GET /v2/entities/{entityId}{?attrs}]
+
++ Parameters
+    + entityId: 'Bcn_Welt' (required, string) - Entity ID
+    + attrs: temperature, humidity (optional, string) - Attributes to be retrieved. Coma separated list.
+
++ Response 200 (application/json)
+
+        {
+            "temperature": 21.7,
+            "humidity": 60,
+        }
+
+### Update or append an entity attribute. POST or PATCH [POST /v2/entities/{entityId}]
+
++ Parameters
+    + entityId: Bcn_Welt (required, string) - Entity ID
+
++ Request (application/json)
+
+    + Headers
+
+            X-HTTP-Method-Override: PATCH
+
+    + Body
+
+            {
+                "ambientNoise": 31.5
+            }
+
++ Response 204
+
+
+### Update or append several entity attributes. POST or PATCH. [PATCH /v2/entities/{entityId}]
+
++ Parameters
+    + entityId: Bcn_Welt (required, string) - Entity ID
+
++ Request (application/json)
+
+        {
+            "temperature": 25.5,
+            "seatsNumber": 6
+        }
+
++ Response 204
+
+
+### Update one or more entity attributes. POST or PATCH [POST /v2/entities/{entityId}{?op}]
+
++ Parameters
+    + entityId: Bcn_Welt (required, string) - Entity ID
+    + op: update - Indicates that an update must be forced
```

Comment:
As far as I understand, the operation to update (that fails if the attributes to update don't exist) is PATCH itself. Thus, `op=update` is redundant.

What I think we need is `op=append` for POST in order to force an "strict append" semantics, instead of the usual "append or update" that POST have (https://github.com/telefonicaid/fiware-orion/issues/983).